extensions/amp-tiktok/amp-tiktok.md

```diff
@@ -0,0 +1,159 @@
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+-->
+
+# amp-tiktok
+
+<!--
+  If the component is relevant for more than one format and operates differently between these
+  formats, include and filter multiple content blocks and code samples.
+-->
+
+## Usage
+
+The simplest usage of the `amp-tiktok` component has the `width`, `height` and `data-src` attributes. This will display the Tiktok embed inside of an iframe on the page.
+
+```html
+ <amp-tiktok width="325" height="575" data-src="6718335390845095173">
+```
+
+`amp-tiktok` does not support autoplay and requires user interaction to play videos.
+
+### Behavior users should be aware of (optional)
+
+## Avoiding Layout shift
+
+The `width` and `height` attributes determine the initial laid out height of the Tiktok embedded in the page.
+
+However, the Tiktok player has a width of 325px and a height of 575px.
+In order to show the entire height of the video, this component will resize its height to include the full player height. Therefore in order to avoid layout shift it is best to match the `width` and `height` attributes to those values.
+
+```html
+ <amp-tiktok width="325" height="575" data-src="6718335390845095173">
+```
+
+If you choose to set the width and height to values greater than those which are reccomended, the TikTok player will still remain at those dimensions, and the surrounding space will be empty.
+
+## Attributes
+
+### `width`
+
+The `width` attribute represents the width of the `amp-tiktok` element.
+However the `iframe` child element will be set to '325px'.
+
+### `height`
+
+The `height` refers to the size of the `amp-tiktok` element. However the Tiktok player is '575px' tall. If the `height` that is set is less than '575px' then the element will resize to fit that new height. This will cause layout shift.
+
+In order to prevent layout shifting, the `height` should set to be at least '575px'.
+
+### `data-src`
+
+The `data-src` attribute can contain one of two values. Either it can contain the video-id or the full source url for the TikTok.
+
+Example with video-id
+
+```html
+ <amp-tiktok width="325" height="575" data-src="6718335390845095173">
+```
+
+Example with source url:
+
+```html
+    <amp-tiktok
+      width="325 "
+      height="575"
+      data-src="https://www.tiktok.com/@scout2015/video/6718335390845095173"
+    >
+```
+
+In liu of the `data-src` attribute, `amp-tiktok` also allows the use of the TikTok generated embed code to display.
+
+To use this method copy the blockquote from the TikTok Embed code:
+
+1. Go to the desktop page of the TikTok you wish to embed
+2. Click the embed link icon (`</>`)
+3. Copy the embed code and add it as a child element of the `amp-tiktok` element.
+4. Add the `placeholder` attribute to the `blockquote`.
+5. Remove the `style` attribute from the `blockquote`.
+
+```html
```

Comment:
This is indented weird, [run it through prettier](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEAeAhgWwA4FoYCWA1jBEQAQDuBAJjABYC8AOiAMwAMHr59cBAc3owWIABxdWAPmZRy8hagBGAGwhgiARwCuEeLIWHy2FejBx6EFTTgAnA0flhTAZxejCJMrjiYlcGlYHRzACeFFhGGwXJAB6WMpEgDpPUiIkyExYgAEXSG0YACYOAEYAVliAN1o4CFiANgB2ErE2NjK2AE4OMQAWMo5OspLGtiC5R3IadBh0XGqbCFxaUSaWto7uvoGhkbGQYKMXGABPFThRTHQAD1xqOnokcnqOMuxrgG5yTAIoO9oGE82IU3tdxo4ZBNHKgXHAwIRoJDJop0IdkbNbAI4CJWAB9VToKBEcHIhSEGDnUS5fJFUplEmk3i2OAAMwiMCiMXiiUoKWIaQyECy1IgBWK5QZpKkIrFdNQsVRUMmSNJqGwKsZAGUwLYsKo4ORtNhDbZyFAsAaAGTkACSgEwCFQqcgwWwnZ0QcgCbRwNzkMKAXg3ALB7gBNyQDwf2jJhgI+iwpTWCyIK7Rdh0FjJYyMVicSB8aYienSfRmWzWJFonEEslUmRBVlZgJYgmk0bU3AC8ipABiJsnZOt+WKxnyDWqwdD+TkuMgbDYlwQFnV4kHJUZ9CY7GiXOEpfRyZF1nszkVnl8rzpTKxeuxGcwOcL-lkdvKzs3u+Lge7wwj5FRlekycXKw6A+gw2IEGAT6OJmG54gS+bLuO8j7iWIBllyla8outaXqmCogXwhAQQhiFdsBxwEeBH5-hCn7kPK6q0b+47QdmW7wbRE6xoBICADZk5CJoIvzoE6c7aFANDkLgzoECyLKQQoyGHuW3JVg+55CrEmDaC44GxAJAhCSouCieJuD1PUYidBIvTmaUzQ9CCpTycOfH6YZ5AmRJUmELJVGMt+iixLC8IEIiEbyqo6haLo8AjgOOD4GpkIgAANCAEDYAiUAuMgoBrrYECUAACmuCA5SgwmUOgJw5WlSi6ho2KaimoRQAIyAut6aW+P4NA2DQAAyhJeq2ABiiZXByvztRVBQQKlqEwJgKgAOr0GEPotXAmplWEBDVKcyDgG4C2-LCtgwEVuoCFcyAssJsJpQAVi41wAEINUQTUWgNvxwHdD1wM9r2atN5wAIoxf9SD3Soj3Tmu51HcIy0Ldgti-DAK0AvQyASGl6MQLCK26tgR3oz6diVP9aU6HocBXRl5UgOgLi4FAcABAEC3MjoBDMldqa3TDgNpbCPwdbYXUgDpbUQ1DANw0DICzEo2MPMghRpS66AECo00AMJCsLIA+mUC3aXAAAq6BKOVsPw5U3o2uJCAwNqGOZQAguJmqnOciuwgAvkHQA).